### DISCUSSION TITLE
Birt 4.13.0: EmitterID org.eclipse.birt.report.engine.emitter.pdf for render option is invalid.

### DISCUSSION THREAD

**Discussion by mr0reed137** [eclipse-birt/birt#1357] [Q&A] [answered]
Hello,
i have an javaee-application (java 11, build-management is done via maven) where i have integrated Birts report engine (4.8.0). Output is pdf format. That works just fine so far. 
Now after updating to Birt 4.13.0 I get this error-message:
**Birt 4.13.0: EmitterID org.eclipse.birt.report.engine.emitter.pdf for render option is invalid.**
I haven't changed any code which looks like this:
```

final IReportEngine reportEngine = BirtEngine.getBirtEngine();
      // ...      
      // Get Runnable Birt-Design 
      IReportRunnable design = getBirtDesign(reportEngine, aBirtInput, theBirtConfigdaten);
      
      // Create Birt-Task 
      final IRunAndRenderTask birtTask = reportEngine.createRunAndRenderTask(design);
      
      ByteArrayOutputStream bos = new ByteArrayOutputStream();
      final PDFRenderOption options = new PDFRenderOption();
    
    options.setOutputStream(bos);
    options.setOutputFormat(IRenderOption.OUTPUT_FORMAT_PDF);
        
    options.setEmitterID(IRenderOption.OUTPUT_EMITTERID_PDF);
    birtTask.setRenderOption(pdfRenderOptions);
   
    birtTask.run();
   
    birtTask.close();
```

The emitter-jar and config is there in folder "reportEngine\plugins":
<webapp-path>\WEB-INF\birt\reportEngine\plugins\org.eclipse.birt.report.engine.emitter.config.pdf_4.13.0.v202303021806.jar
<webapp-path>\WEB-INF\birt\reportEngine\plugins\org.eclipse.birt.report.engine.emitter.pdf_4.13.0.v202303021806.jar

And bundle.infos looks also good 
**<webapp-path>\WEB-INF\birt\reportEngine\configuration\org.eclipse.equinox.simpleconfigurator\bundles.info**
`...
org.eclipse.birt.report.engine.emitter.config.odp,4.13.0.v202303021806,plugins/org.eclipse.birt.report.engine.emitter.config.odp_4.13.0.v202303021806.jar,4,false
org.eclipse.birt.report.engine.emitter.config.ods,4.13.0.v202303021806,plugins/org.eclipse.birt.report.engine.emitter.config.ods_4.13.0.v202303021806.jar,4,false
org.eclipse.birt.report.engine.emitter.config.odt,4.13.0.v202303021806,plugins/org.eclipse.birt.report.engine.emitter.config.odt_4.13.0.v202303021806.jar,4,false
org.eclipse.birt.report.engine.emitter.config.pdf,4.13.0.v202303021806,plugins/org.eclipse.birt.report.engine.emitter.config.pdf_4.13.0.v202303021806.jar,4,false
org.eclipse.birt.report.engine.emitter.config.postscript,4.13.0.v202303021806,plugins/org.eclipse.birt.report.engine.emitter.config.postscript_4.13.0.v202303021806.jar,4,false
org.eclipse.birt.report.engine.emitter.config.ppt,4.13.0.v202303021806,plugins/org.eclipse.birt.report.engine.emitter.config.ppt_4.13.0.v202303021806.jar,4,false
org.eclipse.birt.report.engine.emitter.config.pptx,4.13.0.v202303021806,plugins/org.eclipse.birt.report.engine.emitter.config.pptx_4.13.0.v202303021806.jar,4,false
org.eclipse.birt.report.engine.emitter.config.wpml,4.13.0.v202303021806,plugins/org.eclipse.birt.report.engine.emitter.config.wpml_4.13.0.v202303021806.jar,4,false
org.eclipse.birt.report.engine.emitter.docx,4.13.0.v202303021806,plugins/org.eclipse.birt.report.engine.emitter.docx_4.13.0.v202303021806.jar,4,false
org.eclipse.birt.report.engine.emitter.html,4.13.0.v202303021806,plugins/org.eclipse.birt.report.engine.emitter.html_4.13.0.v202303021806.jar,4,false
org.eclipse.birt.report.engine.emitter.odp,4.13.0.v202303021806,plugins/org.eclipse.birt.report.engine.emitter.odp_4.13.0.v202303021806.jar,4,false
org.eclipse.birt.report.engine.emitter.ods,4.13.0.v202303021806,plugins/org.eclipse.birt.report.engine.emitter.ods_4.13.0.v202303021806.jar,4,false
org.eclipse.birt.report.engine.emitter.odt,4.13.0.v202303021806,plugins/org.eclipse.birt.report.engine.emitter.odt_4.13.0.v202303021806.jar,4,false
org.eclipse.birt.report.engine.emitter.pdf,4.13.0.v202303021806,plugins/org.eclipse.birt.report.engine.emitter.pdf_4.13.0.v202303021806.jar,4,false
org.eclipse.birt.report.engine.emitter.postscript,4.13.0.v202303021806,plugins/org.eclipse.birt.report.engine.emitter.postscript_4.13.0.v202303021806.jar,4,false
org.eclipse.birt.report.engine.emitter.ppt,4.13.0.v202303021806,plugins/org.eclipse.birt.report.engine.emitter.ppt_4.13.0.v202303021806.jar,4,false
org.eclipse.birt.report.engine.emitter.pptx,4.13.0.v202303021806,plugins/org.eclipse.birt.report.engine.emitter.pptx_4.13.0.v202303021806.jar,4,false
org.eclipse.birt.report.engine.emitter.prototype.excel,4.13.0.v202303021806,plugins/org.eclipse.birt.report.engine.emitter.prototype.excel_4.13.0.v202303021806.jar,4,false
org.eclipse.birt.report.engine.emitter.wpml,4.13.0.v202303021806,plugins/org.eclipse.birt.report.engine.emitter.wpml_4.13.0.v202303021806.jar,4,false
org.eclipse.birt.report.engine.fonts,4.13.0.v202303021806,plugins/org.eclipse.birt.report.engine.fonts_4.13.0.v202303021806/,4,false
org.eclipse.birt.report.engine.odf,4.13.0.v202303021806,plugins/org.eclipse.birt.report.engine.odf_4.13.0.v202303021806.jar,4,false
...`

Is there anyone who has had similar problems? 
Thank you for any hints and/or advice...
Marco

**Comment by wimjongman** [comment, marked as answer]
Marco, don't waste your time with 4.13. Try 4.14 instead which is about to be released. Get it from our downloads.

**Reply by mr0reed137** [reply]
Thanks for your reply. I will check it out. Purely out of curiosity: Is version 4.13.0 known to be buggy?

**Reply by wimjongman** [reply]
Yes.

**Reply by mr0reed137** [reply]
One more question:
The version 4.14.0 will presumably work with java 11 and higher?

**Reply by merks** [reply]
It requires Java 17 or higher.

**Reply by mr0reed137** [reply]
Ok. Thanks. 

Any experience yet if Birt 4.14 will run on Tomcat 10.1? I read in other discussion-thread that Birt 4.13 had the issue that it "works with old Java class namespace instead of working with new Jakarta namespace.".... So i guess these is fixed in 4.14 and should run with Tomcat 10.1?

**Reply by speckyspooky** [reply]
No, BIRT 4.14 runs not with Tomcat 10 but you can use the lates Tomcat 9.0.76 instead.

**Reply by mr0reed137** [reply]
@wimjongman Is 4.12 also known to be buggy? Because here I get same error concerning invalid emitterId. I was hoping to be able to upgrade from 4.8 to 4.12 at least, since 4.14 is about to be released in october at the earliest, as it  seems.

**Reply by wimjongman** [reply]
Buggy is maybe not a good word, but there have been issues identified in 4.13 and previous versions that have been addressed in 4.14. Just try out 4.14.

**Reply by mr0reed137** [reply]
Just a quick reply:
I tested 4.14 and with pdf-emitter everything now seems to be working so far.

**Reply by wimjongman** [reply]
Thanks for the feedback!
